### DISCUSSION TITLE
energymeter value needs refresh for getting actual data

### DISCUSSION THREAD

**Discussion by Nifrabar** [zwave-js/zwave-js-ui#3601] [General]
Everspring AN158 Energy meters are included in my Zwave network. Values are sent using MQTT towards Domoticz.
I found that kWh value is frozen in ZwaveJs app . Manual refresh is required to get actual value.
As I m pretty new to ZwaveJs I presume to overlook a setting. 
Hopefully someone might be able to help me fixing this.
-Bart

**Comment by robertsLando** [comment]
Usually this is configrable using a device setting or it's a device issue. @AlCalzone may have more tips

**Comment by Nifrabar** [comment]
Don't think it's a device issue. Once reverted to OZW no problem af all so imho it has to do with ZwaveJs.

**Reply by robertsLando** [reply]
@Nifrabar OZW used to do many weird things in order to support noncompliant devices

**Reply by Nifrabar** [reply]
In addition I found that in ZwaveJs there is an device event which shows update of Watt value . Whatvalue is constantly displayed as actual value.
I don t see an event showing kWh value is being updated. So I wonder (as far as I understand ) if it might be a bug as I only see actual value after manual pussing refreshbutton.

**Reply by robertsLando** [reply]
Please make a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs), loglevel `debug` and attach it here as a file (drag & drop into the text field).

**Reply by Nifrabar** [reply]
@robertsLando

Currently I included:

[zwave-js-ui-store.zip](https://github.com/zwave-js/zwave-js-ui/files/14248563/zwave-js-ui-store.zip)

5 nodes, all power measuring devices Everspring AN158 these are not updating the kWh values
1 battery supplied radiatorthermostatEurotronics Spirit this one seems to work OKe.
Attached the requested logfiles (hopefully these are the requested ones.
[zwave-js-ui-store.zip](https://github.com/zwave-js/zwave-js-ui/files/14248527/zwave-js-ui-store.zip)
Thanks for your help!
-Bart

**Reply by robertsLando** [reply]
cc @AlCalzone

**Reply by robertsLando** [reply]
@Nifrabar I see a tons of:


```
2024-02-12T17:45:34.687Z CNTRLR   [Node 003] [Meter] value[66049]: metadata updated                 [Endpoint 0]
2024-02-12T17:45:34.691Z CNTRLR   [Node 003] [~] [Meter] value[66049]: 52 => 51                     [Endpoint 0]
2024-02-12T17:45:34.696Z DRIVER « [Node 003] [REQ] [ApplicationCommand]
                                  └─[MeterCCReport]
                                      type:       Electric
                                      scale:      W
                                      rate type:  Consumed
                                      value:      51
                                      time delta: 0 seconds
```

Foreach node in the network, seems the report of the energy are sent from the device around every 30 seconds

**Reply by Nifrabar** [reply]
#robertsLando
Thats's correct. As said earlier Watt value is OK but kWh must be refreshed manual. That's not how it should be. kWh should be updated together with W.

**Reply by robertsLando** [reply]
Oh sorry, I misread the issue. Yeah seems it's not coming at all from the device, maybe OZW enabled poll of that device automatically on his side

**Reply by Nifrabar** [reply]
is there a way to poll the kWh value? If ZwaveJs does an automatic refresh every 30 sec. it's solved.

**Reply by robertsLando** [reply]
It's not the best solution to poll devices values but yeah: https://zwave-js.github.io/zwave-js-ui/#/device-support/polling?id=polling-legacy-z-wave-devices
